src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/JsonObject.cs

```diff
@@ -42,11 +42,38 @@ public static object ConvertFromJson(string input, out ErrorRecord error)
             {
                 throw new ArgumentNullException("input");
             }
+
             error = null;
 #if CORECLR
             object obj = null;
             try
             {
+                // JsonConvert.DeserializeObject does not throw an exception when an invalid array is passed,
+                // so we need to do extra logic to identify this scenario using JArray.Parse(input).
+
+                // If input contains '[' or ']', we would like to analyze it further as these markers for a Json data array.
+                if ((input.IndexOf("[", StringComparison.OrdinalIgnoreCase) >= 0) ||
+                    (input.IndexOf("]", StringComparison.OrdinalIgnoreCase) >= 0))
+                {
+                    // Remove carriage return, new-line and empty space from the string.
+                    string stringToValidate = input.Replace(System.Environment.NewLine, string.Empty);
+                    stringToValidate = stringToValidate.Replace(" ", string.Empty);
+
+                    // After removing all white space and new-lines, we are only interested in
+                    // inputs which start or end with a marker for an array.
+                    if (stringToValidate.StartsWith("[", StringComparison.OrdinalIgnoreCase) ||
+                        stringToValidate.EndsWith("]", StringComparison.OrdinalIgnoreCase))
+                    {
+                        // Use JArray.Parse to see if this is a valid array. When an exception is thrown, this means that
+                        // we are potentially processing partial content -- this would be the case for number 2 below.
+
+                        // A Json input from the pipeline is can be represented in the following two ways:
+                        // 1. Each input to the buffer is a complete Json content. There can be multiple inputs of this format.
+                        // 2. The complete buffer input collectively represent a single JSon format. This is typically the majority of the case.
```

Comment:
The comments here are confusing to me -- what is the `buffer` here? Could you please make the comments easier to understand?